### PR TITLE
Add Real time pricing and Notes - AB#13828 / AB#13831

### PR DESCRIPTION
**Pull request by HarshaClouda**
- Add Real-time pricing and Notes to Wish list lines.

[Workspace](https://newwishlist--alfonsob2b.myvtex.com/account#/my-wishlists)

### CODE REVIEW

File: node/clients/masterdata.ts [hunk 1]
```diff
@@ -1,19 +1,29 @@
 import type { InstanceOptions, IOContext } from '@vtex/api'
-import { JanusClient } from '@vtex/api'
+// import { JanusClient } from '@vtex/api'
+import { ExternalClient } from '@vtex/api'
 
 import { DATA_ENTITY_NAME, FIELDS, SCHEMA } from '../utils/constant'
 import { Wishlist } from '../typings/wishlist'
 
-export default class MasterDataClient extends JanusClient {
+// export default class MasterDataClient extends JanusClient {
```
Comment (Jayendra88) [inline node/clients/masterdata.ts]: Remove all the comments

File: node/resolvers/pricing/getPriceBySkuId.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+) => {
+  const {
+    clients: { pricing },
+  } = ctx
+
+  const { skuId, skuReference, productQuantity } = args
+
+  try {
+
+    console.log('skuId: ', skuId)
```
Comment (Jayendra88) [inline node/resolvers/pricing/getPriceBySkuId.ts]: Remove console logs

**Review by Jayendra88 — CHANGES_REQUESTED** [review]

File: node/clients/masterdata.ts (hunk 1 above)
Comment (HarshaClouda) [reply node/clients/masterdata.ts]: Done

File: node/resolvers/pricing/getPriceBySkuId.ts (hunk 2 above)
Comment (HarshaClouda) [reply node/resolvers/pricing/getPriceBySkuId.ts]: Done

File: manifest.json [hunk 3]
```diff
@@ -54,6 +54,9 @@
     {
       "name": "POWER_USER_DS"
     },
+    {
```
Comment (Jayendra88) [inline manifest.json]: do we need this policy anymore

File: node/clients/masterdata.ts [hunk 4]
```diff
@@ -1,30 +1,29 @@
...
+import type { InstanceOptions, IOContext } from "@vtex/api";
+import { ExternalClient } from "@vtex/api";
+import { DATA_ENTITY_NAME, FIELDS, SCHEMA } from "../utils/constant";
+import { Wishlist } from "../typings/wishlist";
 
-import { DATA_ENTITY_NAME, FIELDS, SCHEMA } from '../utils/constant'
-import { Wishlist } from '../typings/wishlist'
-
-export default class MasterDataClient extends JanusClient {
+export default class MasterDataClient extends ExternalClient {
```
Comment (Jayendra88) [inline node/clients/masterdata.ts]: Why we need to make this a External Client

File: node/clients/masterdata.ts [hunk 5]
```diff
@@ -1,30 +1,29 @@
...
   constructor(context: IOContext, options?: InstanceOptions) {
-    super(context, {
+    super(`https://${context.account}.vtexcommercestable.com.br`, context, {
```
Comment (Jayendra88) [inline node/clients/masterdata.ts]: do we need to make this as External client, JanusClient automatically add this URL.

File: node/clients/masterdata.ts [hunk 6]
```diff
@@ -1,30 +1,29 @@
...
       ...options,
       headers: {
         ...options?.headers,
-        VtexIdclientAutCookie: context.authToken,
-        Accept: 'application/vnd.vtex.pricing.v3+json',
-        'Cache-Control': 'no-cache',
-        'Content-Type': 'application/json',
+        ...{ Accept: "application/vnd.vtex.ds.v10+json" },
+        ...(context.authToken
```
Comment (Jayendra88) [inline node/clients/masterdata.ts]: context.authToken should exist always. so no need to check this null value

File: node/clients/masterdata.ts [hunk 7]
```diff
@@ -1,30 +1,29 @@
...
+          ? { VtexIdclientAutCookie: context.authToken }
+          : null),
       },
-    })
+    });
   }
 
   public async getWishlist(id: string): Promise<Wishlist> {
     try {
```
Comment (Jayendra88) [inline node/clients/masterdata.ts]: we can remove async await and try catch from the client. we don't need to handle errors and return.

File: node/clients/pricing.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import type { InstanceOptions, IOContext } from '@vtex/api'
+import { ExternalClient } from '@vtex/api'
+
+export class Pricing extends ExternalClient {
+    constructor(context: IOContext, options?: InstanceOptions) {
```
Comment (Jayendra88) [inline node/clients/pricing.ts]: Why don't we get the price directly from the product

File: node/resolvers/getWishlist.ts [hunk 9]
```diff
@@ -23,6 +23,7 @@ export const getWishlist = async (
     }
 
     return wishlist
+
   } catch (error) {
```
Comment (Jayendra88) [inline node/resolvers/getWishlist.ts]: we don't need try catch on resolvers

File: node/resolvers/getWishlistsByEmail.ts [hunk 10]
```diff
@@ -27,6 +27,7 @@ export const getWishlistsByEmail = async (
 
     return wishlists || []
   } catch (error) {
+    console.error('get wishlists error: ', error)
```
Comment (Jayendra88) [inline node/resolvers/getWishlistsByEmail.ts]: Remove console logs

File: react/components/Notes.jsx [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+      await updateWishlist({
+        variables: {
+          wishlist: {
+            id: wishlistRef.current.id,
+            products: updatedProducts,
+          },
+        },
+      })
+        .then(() => {
```
Comment (Jayendra88) [inline react/components/Notes.jsx]: we shouldn't use .then with async await. 

```
await updateWishList(...)

showToast(...)

setShow(false)

```
Remove this try catch also

File: react/components/Notes.jsx [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+          showToast("Notes saved successfully!");
+        })
+        .catch((error) => {
+          showToast("Error saving notes!");
+          console.error("Error updating products: ", error);
+        });
+
+      setTimeout(function updateShow() {
```
Comment (Jayendra88) [inline react/components/Notes.jsx]: we don't need this timeout when we use async await correctly.

File: react/components/ProductPriceTotal.jsx [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+const ProductPriceTotal = ({ skuReference = 0, productQuantity = 0, currency = '$' }) => {
+
+  const { refetch, data, error: productError, loading } = useQueryGetProductPrice('reference', skuReference)
+
+  useEffect(() => {
+    refetch()
+  }, [productQuantity])
+
+  const price = (data?.product.items ?? []).find((item) => item)?.sellers?.find((seller) => seller)?.commertialOffer?.Price;
```
Comment (Jayendra88) [inline react/components/ProductPriceTotal.jsx]: we should return price from the hook

File: react/components/ProductPriceTotal.jsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from "react"
+import { Spinner } from 'vtex.styleguide'
+import useQueryGetProductPrice from '../hooks/actions/useQueryGetProductPrice'
+
+import styles from './productPriceTotal.css'
+
+const ProductPriceTotal = ({ skuReference = 0, productQuantity = 0, currency = '$' }) => {
+
+  const { refetch, data, error: productError, loading } = useQueryGetProductPrice('reference', skuReference)
```
Comment (Jayendra88) [inline react/components/ProductPriceTotal.jsx]: we should improve this hook to return price, error and loading state only.

File: react/components/ProductPriceTotal.jsx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  useEffect(() => {
+    refetch()
+  }, [productQuantity])
+
+  const price = (data?.product.items ?? []).find((item) => item)?.sellers?.find((seller) => seller)?.commertialOffer?.Price;
+
+  const handlePrice = () => {
+    if (productQuantity && price) {
+      const lineTotal = productQuantity * price
+      return `${currency} ${price ? parseFloat(lineTotal).toFixed(2) : null}`
```
Comment (Jayendra88) [inline react/components/ProductPriceTotal.jsx]: we should format price using react intl. not like this.

File: react/components/ProductPriceTotal.jsx [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  }
+
+  if (productError) {
```
Comment (Jayendra88) [inline react/components/ProductPriceTotal.jsx]: we should combine these conditions

File: react/components/UnitPrice.jsx [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+import { Spinner } from 'vtex.styleguide'
+import useQueryGetProductPrice from '../hooks/actions/useQueryGetProductPrice'
+
+import styles from './unitPrice.css'
+
+const UnitPrice = ({ skuReference = 0, currency = '$' }) => {
+
+  const { data, error: productError, loading } = useQueryGetProductPrice('reference', skuReference)
+
+  const price = (data?.product.items ?? []).find((item) => item)?.sellers?.find((seller) => seller)?.commertialOffer?.Price;
```
Comment (Jayendra88) [inline react/components/UnitPrice.jsx]: we should do the same thing i explained

File: react/components/notes.css [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+.notesButtonContainer {
+  display: flex;
+  justify-content: flex-end;
+  gap: 1rem;
+  padding-top: 1rem;
+}
+
+.notesSubmitButtonContainer {
```
Comment (Jayendra88) [inline react/components/notes.css:24]: remove empty css classes

File: react/hooks/actions/useQueryGetProductPrice.tsx [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+const useQueryWishlists = (field: string, idValue: number | string) => {
+    const { data, error, loading, refetch } = useQuery(PRODUCT_QUERY, {
+        variables: {
+            identifier: { field: field, value: idValue },
+        },
+        notifyOnNetworkStatusChange: true,
+    })
+
+  return {
+    data,
```
Comment (Jayendra88) [inline react/hooks/actions/useQueryGetProductPrice.tsx]: we should write a function to calculate price inside this hook.

File: node/clients/pricing.ts [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+        super(
+            `https://api.vtex.com/${context.account}/pricing/prices`,
+            context,
+            options
+        )
+    }
+
+    public getPriceBySkuId = async (skuId: number) => {
+
+        this.context.logger.debug({
```
Comment (Jayendra88) [inline node/clients/pricing.ts]: Remove these debug loggers

File: node/clients/pricing.ts [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+            auth: this.context.authToken,
+            url: this.context.host,
+        })
+
+        const priceEndpoint = `/${skuId}`
```
Comment (Jayendra88) [inline node/clients/pricing.ts]: we don't need this varibale

**Review by Jayendra88 — CHANGES_REQUESTED** [review]

File: node/resolvers/pricing/getPriceBySkuId.ts [hunk 22]
```diff
@@ -0,0 +1,28 @@
+export const getPriceBySkuId = async (
+  _: unknown,
```
Comment (Jayendra88) [inline node/resolvers/pricing/getPriceBySkuId.ts]: remove this

File: react/components/UnitPrice.jsx [hunk 23]
```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from "react"
+import { Spinner } from 'vtex.styleguide'
+import useQueryGetProductPrice from '../hooks/actions/useQueryGetProductPrice'
+
+import styles from './unitPrice.css'
+
+const UnitPrice = ({ skuReference = 0, currency = '$' }) => {
+
+  const { data, error: productError, loading } = useQueryGetProductPrice('reference', skuReference)
```
Comment (Jayendra88) [inline react/components/UnitPrice.jsx]: `useProductPrice`

File: react/components/notes.css (hunk 18 above)
Comment (HarshaClouda) [reply react/components/notes.css:24]: Can't remove this. If removed the block class will be undefined.

File: node/clients/masterdata.ts (hunk 4 above)
Comment (HarshaClouda) [reply node/clients/masterdata.ts]: Removed

File: node/clients/masterdata.ts (hunk 5 above)
Comment (HarshaClouda) [reply node/clients/masterdata.ts]: Removed

File: node/clients/masterdata.ts (hunk 6 above)
Comment (HarshaClouda) [reply node/clients/masterdata.ts]: Restored old code

File: manifest.json (hunk 3 above)
Comment (HarshaClouda) [reply manifest.json]: Removed

File: node/clients/masterdata.ts (hunk 7 above)
Comment (HarshaClouda) [reply node/clients/masterdata.ts]: Done

File: node/clients/pricing.ts (hunk 8 above)
Comment (HarshaClouda) [reply node/clients/pricing.ts]: The client is removed.

File: node/clients/pricing.ts (hunk 20 above)
Comment (HarshaClouda) [reply node/clients/pricing.ts]: The client is removed.

File: node/clients/pricing.ts (hunk 21 above)
Comment (HarshaClouda) [reply node/clients/pricing.ts]: The client is removed.

File: node/resolvers/getWishlist.ts (hunk 9 above)
Comment (HarshaClouda) [reply node/resolvers/getWishlist.ts]: Fixed

File: node/resolvers/getWishlistsByEmail.ts (hunk 10 above)
Comment (HarshaClouda) [reply node/resolvers/getWishlistsByEmail.ts]: Removed

File: react/components/Notes.jsx (hunk 11 above)
Comment (HarshaClouda) [reply react/components/Notes.jsx]: Done

File: react/components/Notes.jsx (hunk 12 above)
Comment (HarshaClouda) [reply react/components/Notes.jsx]: Done

File: react/components/ProductPriceTotal.jsx (hunk 14 above)
Comment (HarshaClouda) [reply react/components/ProductPriceTotal.jsx]: Done

File: react/components/ProductPriceTotal.jsx (hunk 13 above)
Comment (HarshaClouda) [reply react/components/ProductPriceTotal.jsx]: Done

File: react/components/UnitPrice.jsx (hunk 17 above)
Comment (HarshaClouda) [reply react/components/UnitPrice.jsx]: Done

File: react/components/ProductPriceTotal.jsx (hunk 15 above)
Comment (HarshaClouda) [reply react/components/ProductPriceTotal.jsx]: Have an issue with doing this change.

File: react/components/ProductPriceTotal.jsx (hunk 16 above)
Comment (HarshaClouda) [reply react/components/ProductPriceTotal.jsx]: Done

File: react/hooks/actions/useQueryGetProductPrice.tsx (hunk 19 above)
Comment (HarshaClouda) [reply react/hooks/actions/useQueryGetProductPrice.tsx]: Done

**Review by Jayendra88 — APPROVED** [review]

**Pull request merged**